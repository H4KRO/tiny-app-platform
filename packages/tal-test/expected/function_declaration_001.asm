main()
  entry:
    FunctionRef       name: f_0
    DeclareLocal      name: f, mutable: false, hasInitialValue: true
    Pop               inBlock: false
    Literal 0
    MakeArray
    Literal "x"
    Literal 2
    Literal 1
    MakeRecord
    Local             name: f
    Call
    Pop               inBlock: false
    Literal 2
    Literal 1
    MakeArray
    Literal 0
    MakeRecord
    Local             name: f
    Call
f_0([object Object])
  entry:
    Local             name: x
    Literal 1
    Intrinsic         operation: INTRINSIC_ADD
