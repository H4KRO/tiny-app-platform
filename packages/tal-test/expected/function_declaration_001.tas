fun f(x) x + 1
f(x: 2)
f(2)
