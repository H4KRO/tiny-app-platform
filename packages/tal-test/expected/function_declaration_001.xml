<program>
  <declarelocal>
    <function>
      <binaryoperator>
        <local />
        <literal />
      </binaryoperator>
    </function>
  </declarelocal>
  <call>
    <local />
    <literal />
  </call>
  <call>
    <local />
    <literal />
  </call>
</program>
